packages/calypso-e2e/src/lib/pages/blaze-campaign-page.ts

```diff
@@ -43,6 +43,8 @@ export class BlazeCampaignPage {
 	 * @param {TestFile} path TestFile object.
 	 */
 	async uploadImage( path: TestFile ) {
+		await this.page.getByRole( 'region', { name: 'Appearance' } ).waitFor( { timeout: 15 * 1000 } );
```

Comment:
I will run another test just in case, but I am unsure if that change will work fine. 

The flaky test happens because the timeout occurs in the `this.page.waitForEvent( 'filechooser' );` promise.
The problem is that we need to give the page enough time to finish loading all the steps and click the Upload button before that timeout hits. 

That is why I separated the wait for the page, and the upload image in two lines. I thought that this would be easier to follow if another person read it.

From my understanding (that is a bit lacking), I need to first start the promise for the filechooser, and then activate the element that uses that file chooser. The second element needs to be executed before the first promise timeouts. 

I can change it like you mentioned there, but I will need to increase the timeout of both lines (fileChooser and click), to give them enough time for both to finish correctly.

@WunderBart, let me know if you want me to apply that change 👍

Reply:
Thanks for the context! It helps a lot to understand the issue.

> I can change it like you mentioned there, but I will need to increase the timeout of both lines (fileChooser and click), to give them enough time for both to finish correctly.

In this case, it sounds like it would suffice to only apply the extended timeout to the `filechooser` event and let the `Upload` timeout with the default value:

```js
const fileChooserPromise = this.page.waitForEvent( 'filechooser', { timeout: 15_000 } );

await this.page
	.getByRole( 'region', { name: 'Appearance' } )
	.getByRole( 'button', { name: 'Upload' } )
	.click();

const fileChooser = await fileChooserPromiser;
```

Reply:
Nice, I will try that then.